File: games/Starfall/Script/TargetProfile.lua
-- [ts]: TargetProfile.ts
local ____exports = {} -- 1
local ____Dora = require("Dora") -- 1
local App = ____Dora.App -- 1
local Size = ____Dora.Size -- 1
function ____exports.desktopProfile() -- 13
	return { -- 14
		kind = "desktop", -- 15
		width = 960, -- 16
		height = 640, -- 17
		showVirtualPad = false, -- 18
		controls = "WASD / ARROWS MOVE   MOUSE AIM   CLICK / SPACE DASH   P PAUSE   R RESTART" -- 19
	} -- 19
end -- 13
function ____exports.handheldProfile() -- 23
	return { -- 24
		kind = "handheld", -- 25
		width = 640, -- 26
		height = 480, -- 27
		showVirtualPad = App.platform == "macOS" or App.platform == "Windows", -- 28
		controls = "LEFT STICK / DPAD MOVE   RIGHT STICK AIM   A DASH   START PAUSE   X RESTART" -- 29
	} -- 29
end -- 23
function ____exports.applyTargetProfile(profile) -- 33
	App.targetFPS = 60 -- 34
	if App.platform == "macOS" or App.platform == "Windows" then -- 34
		App.fullScreen = false -- 36
		App.winSize = Size(profile.width, profile.height) -- 37
	end -- 37
end -- 33
return ____exports -- 33

File: games/Starfall/Script/TargetProfile.ts
import { App, PlatformType, Size } from "Dora";

export type TargetKind = "desktop" | "handheld";

export interface TargetProfile {
	kind: TargetKind;
	width: number;
	height: number;
	showVirtualPad: boolean;
	controls: string;
}

export function desktopProfile(): TargetProfile {
	return {
		kind: "desktop",
		width: 960,
		height: 640,
		showVirtualPad: false,
		controls: "WASD / ARROWS MOVE   MOUSE AIM   CLICK / SPACE DASH   P PAUSE   R RESTART",
	};
}

export function handheldProfile(): TargetProfile {
	return {
		kind: "handheld",
		width: 640,
		height: 480,
		showVirtualPad: App.platform === PlatformType.macOS || App.platform === PlatformType.Windows,
		controls: "LEFT STICK / DPAD MOVE   RIGHT STICK AIM   A DASH   START PAUSE   X RESTART",
	};
}

export function applyTargetProfile(profile: TargetProfile): void {
	App.targetFPS = 60;
	if (App.platform === PlatformType.macOS || App.platform === PlatformType.Windows) {
		App.fullScreen = false;
		App.winSize = Size(profile.width, profile.height);
	}
}
